cmd/frcli/fiat_estimate.go

```diff
@@ -67,24 +67,30 @@ func queryFiatEstimate(ctx *cli.Context) error {
 		return err
 	}
 
-	var customPrices []*frdrpc.BitcoinPrice
+	// nolint: prealloc
+	var filteredPrices []*frdrpc.BitcoinPrice
 
 	if fiatBackend == frdrpc.FiatBackend_CUSTOM {
-		customPrices, err = parsePricesFromCSV(
+		customPrices, err := parsePricesFromCSV(
 			ctx.String("prices_csv_path"),
 			ctx.String("custom_price_currency"),
 		)
 		if err != nil {
 			return err
 		}
+
+		filteredPrices, err = filterPrices(customPrices, ts, ts)
+		if err != nil {
+			return err
+		}
 	}
 
 	// Set start and end times from user specified values, defaulting
 	// to zero if they are not set.
 	req := &frdrpc.ExchangeRateRequest{
 		Timestamps:   []uint64{uint64(ts)},
 		FiatBackend:  fiatBackend,
-		CustomPrices: customPrices,
+		CustomPrices: filteredPrices,
 	}
 
 	rpcCtx := context.Background()
```

Comment:
ok so technically now with this filtering (in the case of Custom price source backend) only 1 timestamp will be sent to faraday and will be returned as is .... so for fiat_estimate (given custom option) faraday doesnt do anything... so the call to faraday could just be skipped here and all processing can be done in frcli (even in the case of multiple timestamps). Then the `CustomPrices` field in the `ExchangeRateRequest` can also be removed i think. Let me know what you think . The upside, i think, is an unnecessary network call and the downside is some code bloat and possible duplicate logic (in the case of multiple timestamps)

Reply:
I think that it's ok to leave the call in. There's a use case for throwing a ton of custom price data at faraday and wanting it to pick the right timestamp + do duplication. I'd err on the side of not adding custom logic to save ourselves one rpc call.